src/Player/PlayerState/PlayerStateHack.cpp

```diff
@@ -0,0 +1,276 @@
+        al::copyPose(getActor(), mHackKeeper->getHack());
+        if (!mHackKeeper->getHackSensor() || !mHackKeeper->isHackDemoStarted()) {
+            mHackKeeper->killHackDemoModel();
+            mHackKeeper->recordHack();
+            mHackCap->recordHack();
+            al::startHitReaction(mHackCap, "ひょうい完了");
+            al::setNerve(this, &nrvPlayerStateHack.Hack);
+        }
+    }
+
+}
+
+void PlayerStateHack::exeHackDemoPuppetable() {
+    auto actor = getActor();
+    if (al::isFirstStep(this)) {
+        mAnimator->startAnim("Wait");
+        mModelChanger->hideModel();
+        al::offCollide(actor);
+        al::setVelocityZero(actor);
+        al::tryKillEmitterAndParticleAll(actor);
+    }
+    if (!mHackKeeper->getHackSensor() || !mHackKeeper->isHackDemoStarted()) {
+        mHackCap->noticeHackDemoPuppetableEnd();
+        mHackKeeper->recordHack();
+        mHackCap->recordHack();
+        al::setNerve(this, &nrvPlayerStateHack.Hack);
+    }
+}
+
+bool PlayerStateHack::isEnableCancelHack() const {
+    if (al::isNerve(this, &nrvPlayerStateHack.Hack) && (!mHackKeeper->getHackSensor() || !mHackKeeper->isStartedHacking())) {
+        return al::isGreaterEqualStep(this, 13);
+    }
+    return false;
+}
+
+bool PlayerStateHack::isEnableChangeState() const {
+    auto hackKeeper = mHackKeeper;
+    if (hackKeeper->isHack() || (hackKeeper->getHackSensor() && hackKeeper->isStartedHacking()) || al::isNerve(this, &nrvPlayerStateHack.HackDemo)) {
+        return false;
+    } else {
+        return !al::isNerve(this, &nrvPlayerStateHack.HackDemoPuppetable);
+    }
+}
+
+bool PlayerStateHack::isEnableModelSyncShowHide() const {
+    if (al::isNerve(this, &nrvPlayerStateHack.Hack))
+        return mHackKeeper->isHack();
+    else
+        return true;
+}
+
+bool PlayerStateHack::isIgnoreUpdateCollider() const {
+    if (isDead())
+        return true;
+    return !al::isNerve(this, &End);
+}
+
+void PlayerStateHack::prepareEndHack() {
+    al::startHitReaction(getActor(), "ひょうい終了");
+    CapFunction::endHack(mHackCap, mAnimator);
+    al::onCollide(getActor());
+    mAnimator->startAnim("JumpEndHack");
+    mModelChanger->showModel();
+    mHackKeeper->killHackDemoModel();
+    al::tryDeleteEffect(getActor(), "PossessTrace");
+    al::setNerve(this, &End);
+}
+
+void PlayerStateHack::prepareStageStartHack() {
+    auto actor = getActor();
+    mIsStageStartHack = true;
+    al::offCollide(actor);
+    al::setVelocityZero(actor);
+    al::tryKillEmitterAndParticleAll(actor);
+    mModelChanger->hideModel();
+    mHackKeeper->recordHack();
+    mHackCap->recordHack();
+}
+
+void PlayerStateHack::prepareStartHack(const al::HitSensor* source, const al::HitSensor* target) {
+    mHackCap->startHack();
+    sead::Vector3f upDir;
+    al::calcUpDir(&upDir, getActor());
+    f32 sensorDistance = al::calcDistance(source, target);
+    sensorDistance -= 200.0f;
+    sensorDistance = (sensorDistance < 0.0f ? 0.0f : sensorDistance)/30.f;
```

Comment:
```suggestion
    sensorDistance = sead::Mathf::max(sensorDistance, 0)/30.0f;
```

Reply:
This could probably even inline the line above then, as I don't think N would do so many operations on a single local variables in a row... it's probably just a asm/decompiler artifact.